### PR TITLE
test: add headerslist copy check

### PR DESCRIPTION
**Pull request by tsctx**
cc @KhafraDev

### CODE REVIEW

File: test/fetch/request.js [hunk 1]
```diff
@@ -465,12 +465,13 @@ test('request.referrer', () => {
 
 // https://github.com/nodejs/undici/issues/2445
 test('Clone the set-cookie header when Request is passed as the first parameter and no header is passed.', (t) => {
-  const { strictEqual } = tspl(t, { plan: 2 })
   const request = new Request('http://localhost', { headers: { 'set-cookie': 'A' } })
   const request2 = new Request(request)
+  assert.strictDeepEqual([...request.headers], [['set-cookie', 'A']])
```
Comment (tsctx) [inline test/fetch/request.js:470]:
```suggestion
  assert.deepStrictEqual([...request.headers], [['set-cookie', 'A']])
```

File: test/fetch/request.js [hunk 2]
```diff
@@ -465,12 +465,13 @@ test('request.referrer', () => {
...
   request2.headers.append('set-cookie', 'B')
-  strictEqual(request.headers.getSetCookie().join(', '), request.headers.get('set-cookie'))
-  strictEqual(request2.headers.getSetCookie().join(', '), request2.headers.get('set-cookie'))
+  assert.strictDeepEqual([...request.headers], [['set-cookie', 'A']])
```
Comment (tsctx) [inline test/fetch/request.js:472]:
```suggestion
  assert.deepStrictEqual([...request.headers], [['set-cookie', 'A']])
```

**Comment by KhafraDev** [comment]
Can you also add a test that for when `headers` is passed as an option to RequestInit?

File: test/fetch/request.js [hunk 3]
```diff
@@ -449,7 +449,8 @@ test('set-cookie headers get cleared when passing a Request as first param', ()
 
   assert.deepStrictEqual([...req1.headers], [['set-cookie', 'a=1']])
   const req2 = new Request(req1, { headers: {} })
-
+  assert.deepStrictEqual([...req1.headers], [['set-cookie', 'a=1']])
+
```
Comment (tsctx) [inline test/fetch/request.js:453]:
```suggestion

```

**Comment by tsctx** [comment]
Added

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**